lychee-bin/src/options.rs

```diff
@@ -528,6 +537,33 @@ with a status code of 429, 500 and 501."
     #[serde(default = "max_concurrency")]
     pub(crate) max_concurrency: usize,
 
+    /// Default maximum concurrent requests per host (default: 10)
+    ///
+    /// This limits the maximum amount of requests that are sent simultaneously
+    /// to the same host. This helps to prevent overwhelming servers and
+    /// running into rate-limits. Use the `hosts` option to configure this
+    /// on a per-host basis.
+    ///
+    /// Examples:
+    ///   --host-concurrency 2   # Conservative for slow APIs
+    ///   --host-concurrency 20  # Aggressive for fast APIs
+    #[arg(long = "host-concurrency", verbatim_doc_comment)]
+    #[serde(default)]
+    pub(crate) host_concurrency: Option<usize>,
+
+    /// Minimum interval between requests to the same host (default: 100ms)
+    ///
+    /// Sets a baseline delay between consecutive requests to prevent
+    /// hammering servers. The adaptive algorithm may increase this based
```

Comment:
My choice of wording here wasn't perfect. Maybe we should replace the term "hammering" with "overloading"?


```suggestion
    /// overloading servers. The adaptive algorithm may increase this based
```